### PR TITLE
BRS for Azure_ServiceBus_AuthN_Disable_Local_Auth

### PR DESCRIPTION
**Pull request by cbulbule**
BRS for Azure_ServiceBus_AuthN_Disable_Local_Auth

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1 [hunk 1]
```diff
@@ -0,0 +1,800 @@
...
+           Disable-AzServiceBusLocalAuth -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        2. To disable local authentication on of all Service Bus Namespaces in a Subscription:
+           Disable-AzServiceBusLocalAuth -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        3. To disable local authentication on of all Service Bus Namespaces in a Subscription, from a previously taken snapshot:
+           Disable-AzServiceBusLocalAuth -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\Disable-LocalAuthForServiceBusNamespaces\ServiceBusWithoutLocalAuthDisabled.csv
+
+        To know more about the options supported by the remediation command, execute:
+        Get-Help Disable-BasicAuthForAppServices -Detailed
```
Comment (MJaithra) [inline Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1:50]: App service -> Service Bus Namespaces

File: Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1 [hunk 2]
```diff
@@ -0,0 +1,800 @@
...
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage = "Specifies a dry run of the actual remediation")]
+        $DryRun,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage = "Specifies the path to the file to be used as input for the remediation")]
+        $InputFilePath,
+
+        [Switch]        
+        [Parameter(ParameterSetName = "WetRun", HelpMessage = "Specifies script is run as a subroutine of AutoRemediation Script")]
```
Comment (MJaithra) [inline Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1:186]: please check helpMessage once

File: Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1 [hunk 3]
```diff
@@ -0,0 +1,800 @@
...
+                $userInput = Read-Host -Prompt "(Y|N)"
+
+                if ($userInput -ne "Y") {
+                    Write-Host "Local authentication will not be changed for Service Bus Namespace(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                    Write-Host $([Constants]::DoubleDashLine)	
+                    return
+                }
+            }
+            else {
+                Write-Host "'Force' flag is provided. Disable local authentication will be enabled on Service Bus Namespace(s) in the Subscription without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
```
Comment (MJaithra) [inline Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1:453]: Please see if this is more meaningful. "Disable local authentication will be enabled on Service Bus Namespace(s)" -> "Local auth will be disbaled on Service Bus Namespace(s)"

File: Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1 [hunk 4]
```diff
@@ -0,0 +1,800 @@
...
+                Write-Host "The information related to Service Bus Namespace(s) where disable local authentication was not set, has been saved to [$($serviceBusNamespacesSkippedFile)]. Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+
+        }
+         else {
+
+            Write-Host "Remediation Summary: " -ForegroundColor $([Constants]::MessageType.Info)
+            if ($($serviceBusNamespacesRemediated | Measure-Object).Count -gt 0) {
+                Write-Host "Successfully set disable lLocal authentication to true for the following Service Bus Namespaces(s) in the subscription:" -ForegroundColor $([Constants]::MessageType.Update)
```
Comment (MJaithra) [inline Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1:525]: NIT : typo Local authentication

File: Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1 (hunk 1 above)
Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1:50]: Done

File: Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1 (hunk 2 above)
Comment (cbulbule) [reply Scripts/RemediationScripts/Remediate-DisableBasicAuthForServiceBusNamespaces.ps1]: done

**Review by MJaithra — APPROVED** [review]

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**